apollo-router/src/services/layers/apq.rs

```diff
@@ -244,6 +244,15 @@ mod apq_tests {
             .get(CACHE_CONTROL)
             .is_none());
 
+        // We need to yield here to make sure the router
+        // has had enough time to insert the full query into the apq cache.
```

Comment:
the yield is not there to give enough time to insert, but to let run the cleanup task for the entry returned by `get()`

Reply:
hmm i gotta update the comment

Reply:
fixed in [21b98c9](https://github.com/apollographql/router/pull/2519/commits/21b98c994070c3c9f1a5383285a11ce6489930d5)